src/components/Card/Card.js

```diff
@@ -0,0 +1,70 @@
+import PropTypes from 'prop-types'
+import './card.css'
+
+function Card({ jobData }) {
+  const isDataPresent = field => {
+    return typeof field === 'string' ? field : false
+  }
+
+  const removeHTML = text => {
+    return text.replace(/<[^>]*(>|$)|&nbsp;|&zwnj;|&raquo;|&laquo;|&bull;|&gt;/g, '')
+  }
+
+  const isValidURL = string => {
+    return !string
+      ? false
+      : string.match(
+          // eslint-disable-next-line no-useless-escape
+          /^(?:(?:(?:https?|ftp):)?\/\/)(?:\S+(?::\S*)?@)?(?:(?!(?:10|127)(?:\.\d{1,3}){3})(?!(?:169\.254|192\.168)(?:\.\d{1,3}){2})(?!172\.(?:1[6-9]|2\d|3[0-1])(?:\.\d{1,3}){2})(?:[1-9]\d?|1\d\d|2[01]\d|22[0-3])(?:\.(?:1?\d{1,2}|2[0-4]\d|25[0-5])){2}(?:\.(?:[1-9]\d?|1\d\d|2[0-4]\d|25[0-4]))|(?:(?:[a-z\u00a1-\uffff0-9]-*)*[a-z\u00a1-\uffff0-9]+)(?:\.(?:[a-z\u00a1-\uffff0-9]-*)*[a-z\u00a1-\uffff0-9]+)*(?:\.(?:[a-z\u00a1-\uffff]{2,})))(?::\d{2,5})?(?:[/?#]\S*)?$/i
+        )
+  }
+
+  const title = removeHTML(isDataPresent(jobData.title))
+  const company = removeHTML(isDataPresent(jobData.company.display_name))
+  const location = removeHTML(isDataPresent(jobData.location.display_name))
+  const description = removeHTML(isDataPresent(jobData.description))
+  const redirectUrl = isValidURL(isDataPresent(jobData.redirect_url))
+  const date = Date.parse(removeHTML(jobData.created))
+    ? new Date(removeHTML(jobData.created)).toLocaleDateString()
+    : false
+  const valid = date && title && company && location && description && redirectUrl ? true : false
```

Comment:
Is this check needed, what is the purpose of `isDataPresent`, can valid be removed all-together?

Reply:
I think it was added because we are kicking out the cards that don't have all the fields on it i.e. partial cards

Reply:
Yes, If we don't have a valid posted date to show the user then we boot the card.